src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -424,12 +427,14 @@ public synchronized void handleMessageLayoutPropose(LayoutRankMsg msg, ChannelHa
     // TODO How do we handle holes in history if let in layout commit message. Maybe we have a hole filling process
     // TODO how do reject the older epoch commits, should it be an explicit NACK.
     public synchronized void handleMessageLayoutCommit(LayoutRankMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        Layout commitLayout = msg.getLayout();
+
         long serverEpoch = getServerEpoch();
-        if(msg.getLayout().getEpoch() <= serverEpoch) {
+        if(commitLayout.getEpoch() < serverEpoch) {
```

Comment:
correct to change from msg.getLayout() to commitLayout.getEpoch() ; same explanation as above.

I might add:
 
   if (commitLayout.getEpoch() == serverEpoch)
       // do nothing, just ACK